### PR TITLE
[Fortran/gfortran] Sync gfortran tests with upstream

### PR DESCRIPTION
**Pull request by tarunprabhu**
These are synced with [a9e9f772c7488](https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=a9e9f772c7488ac0c09dd92f28890bdab939771a).

The static test configuration files, and denylists have been updated. A quick fix is also included that prevents ninja builds from being serialized.

-------------------------------
The number of files that have been changed are enormous. However, the overwhelming majority of these are either new files that have been added to the gfortran testsuite, or updates to existing ones so they are identical to upstream. Another dozen or so files are the static test configuration files that are generated with `Fortran/gfortran/utils/update-test-config.py`. 

The files below are those that do need to be reviewed:

- [`gfortran/CMakeLists.txt`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-f85fabb8751d614f90ebd72b4d39a3c90b40f092b9c55ff20c774918d5565124)
- [`gfortran/regression/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-eb0bee91f1aa9ac617939b2e138cb659da256de8675932e3ed73647dd3cff26f)
- [`gfortran/regression/coarray/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-b6d2f7882f25c082f1268c10e868866dc366f8b715109df2b4e32cf4508cbcee)
- [`gfortran/regression/goacc/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-77548f2534fda53e6cacf2ff7b2abf023b7a95dea5b49477ceb90d0f552cfef1)
- [`gfortran/regression/gomp/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-71008dfb3f27d5cd59f3a75b7b65ee496f0a3f2c2aab8a56be389fd7ee0ab03d)
- [`gfortran/regression/ieee/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-a6871a3412d76764e6437bd625551684ab45e3597456c415d701779ad799e678)
- [`gfortran/regression/ubsan/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-efb6ace0ec6cbe4af96498d2daddc5bbc0fe672273a4344232a2513fc3a3dffc)
- [`gfortran/regression/vect/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-b6b9f4d68e354ac5c79c6d5644aad4d21d61c8ab01da70f0d16864cf25b2efff)
- [`gfortran/torture/execute/DisabledFiles.cmake`](https://github.com/llvm/llvm-test-suite/pull/151/files#diff-a386d42a3d82dba52c4c4621e1bfdb379bda290da3f8fac5acf8141efdf8ae9a)

### CODE REVIEW

**Review by luporl — COMMENTED** [review]
@tarunprabhu I've checked the files that you indicated that need review and they look good to me.

But I've run the testsuite on AArch64 and got 2 failures:
```
  test-suite :: Fortran/gfortran/regression/gfortran-regression-execute-regression__ishftc_optional_size_1_f90.test
  test-suite :: Fortran/gfortran/regression/gfortran-regression-execute-regression__pr99210_f90.test
```

The first test causes a segmentation fault and the second exits with `ERROR STOP: code 2`.

BTW, thanks for your work in updating the gfortran tests!

**Comment by tarunprabhu** [comment]
Thanks for checking, Leandro. 

I have added the two failing tests to the denyilsts. I'll leave this up for a bit longer to let the others test it out on a few more platforms.

File: Fortran/gfortran/regression/DisabledFiles.cmake [hunk 1]
```diff
@@ -890,6 +890,9 @@ file(GLOB SKIPPED_FILES CONFIGURE_DEPENDS
   # with procedure designator 'new_t' with explicit interface that cannot be
   # called via an implicit interface
   pr112407a.f90
+
+  # This causes a segmentation fault at compile-time.
```
Comment (luporl) [inline Fortran/gfortran/regression/DisabledFiles.cmake:894]: Actually, the segmentation fault happens at run-time.

Comment (tarunprabhu) [reply Fortran/gfortran/regression/DisabledFiles.cmake]: Fixed. Thanks for the clarification.

**Review by luporl — APPROVED** [review]

**Comment by tblah** [comment]
Thanks for this update!

I have one more failure:
```
Failed Tests (1):
  test-suite :: Fortran/gfortran/regression/gfortran-regression-compile-regression__submodule_3_f08.test
```

**Comment by tarunprabhu** [comment]
>   test-suite :: Fortran/gfortran/regression/gfortran-regression-compile-regression__submodule_3_f08.test
> ```

Added to disabled tests.

**Review by jeanPerier — APPROVED** [review]
Thanks for this update! It worked for me:

```
Total Discovered Tests: 6598
  Passed: 6598 (100.00%)
```

**Comment by AlexisPerry** [comment]
I just ran this on an amd-rome node and all passed for me.  :-)

**Review by AlexisPerry — APPROVED** [review]

**Review by tblah — APPROVED** [review]
All works for me now. Thanks for the update!

**Comment by tarunprabhu** [comment]
Thanks @AlexisPerry, @jeanPerier, @luporl and @tblah for the reviews and checking that it worked.

**Pull request merged**